homeassistant/components/sonos/speaker.py

```diff
@@ -1032,7 +1032,7 @@ def _restore_groups(
 
             return groups
 
-        def _restore_players(speakers: list[SonosSpeaker]) -> None:
+        def _restore_players(speakers: Collection[SonosSpeaker]) -> None:
```

Comment:
Just curious: what does this typing do for us that the previous one didn't?

Reply:
Both `_restore_players` and `_restore_groups` are actually called with sets. That was probably missed during a refactoring. The `hass.async_add_executor_job` call isn't validated yet. Mypy is still missing some typing features for it.

_I do however have a WIP branch with some workarounds which is how I found it. If you're interested: https://github.com/home-assistant/core/compare/dev...cdce8p:ha-core:type-core-HassJob-ParamSpec_
https://github.com/home-assistant/core/blob/13d3f4c3b2ec7f2a40c8316e7bf41d3148582d44/homeassistant/components/sonos/speaker.py#L1060